plugins/reach/plugins/OpenCalais/batch/kReachVendorTaskOpenCalaisProcessorEngine.php

```diff
@@ -295,15 +294,18 @@ protected function retrieveAttachmentAssetContent ($assetId)
 
     protected function sendOpenCalaisGetTagsRequest ($transcript, $partnerId)
     {
+	    $openCalaisData = $this->getOpenCalaisSimpleXMLElementFromPartnerMetadata($partnerId);
+	    $url = $this->retrieveValueFromXml($openCalaisData, self::OPEN_CALAIS_API_URL_FIELD_NAME);
+
```

Comment:
@hilak My suggestion is to have here condition: if(empty($url)){throw exception/write to log} - will happen if no url was set in KMS

Reply:
@TalDubovKaltura I will add it, though the same can be achieved by making the endpoint field mandatory at the XSD level.